src/tree.ml

```diff
@@ -495,10 +504,11 @@ module CM = struct
         in
         let env_builder =
           match i with
-          | Some(i) -> (slot, (i, Array.map to_tvar e)) :: r.env_builder
+          | Some(i) ->
+              (slot, (i, Array.map (index_var vi) e)) :: r.env_builder
           | None    -> r.env_builder
         in
-        { r with env_builder ; cond_pool }
+        {r with env_builder; cond_pool}
     | _             -> r
 
   (** [specialize pat col pos cl] filters and transforms LHS of [cl] assuming
```

Comment:
It would be nice that in all these functions (specialize, abstract, etc.) the arguments name in the comment and in the code are the same.